examples/basic_operations/add_responsive_search_ad.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This example adds a responsive search ad to a given ad group.
+
+To get ad groups, run get_ad_groups.py.
+"""
+
+
+import argparse
+import sys
+from uuid import uuid4
+
+from google.ads.google_ads.client import GoogleAdsClient
+from google.ads.google_ads.errors import GoogleAdsException
+
+
+def main(client, customer_id, ad_group_id):
+    ad_group_ad_service = client.get_service('AdGroupAdService', version='v2')
+    ad_group_service = client.get_service('AdGroupService', version='v2')
+
+    pinned_headline = client.get_type('AdTextAsset', version='v2')
+    pinned_headline.text.value = f'Cruise to Mars #{str(uuid4())[:8]}'
+    # Set a pinning to always choose this asset for HEADLINE_1. Pinning is
+    # optional; if no pinning is set, then headlines and descriptions will be
+    # rotated and the ones that perform best will be used more often.
+    pinned_headline.pinned_field = client.get_type(
+       'ServedAssetFieldTypeEnum', version='v2').HEADLINE_1
+
+    # Create the ad group ad
+    ad_group_ad_operation = client.get_type('AdGroupAdOperation', version='v2')
+    ad_group_ad = ad_group_ad_operation.create
+    ad_group_ad.status = client.get_type(
+        'AdGroupAdStatusEnum', version='v2').PAUSED
+    ad_group_ad.ad_group.value = ad_group_service.ad_group_path(
+        customer_id, ad_group_id)
+
+    # Set responsive search ad info
+    final_url = ad_group_ad.ad.final_urls.add()
+    final_url.value = 'http://www.example.com'
+    ad_group_ad.ad.responsive_search_ad.headlines.extend([
+        pinned_headline,
+        _create_ad_text_asset(client, 'Best Space Cruise Line'),
+        _create_ad_text_asset(client, 'Experience the Stars')])
+    ad_group_ad.ad.responsive_search_ad.descriptions.extend([
+        _create_ad_text_asset(client, 'Buy your tickets now'),
+        _create_ad_text_asset(client, 'Visit the Red Planet')])
+    ad_group_ad.ad.responsive_search_ad.path1.value = 'all-inclusive'
+    ad_group_ad.ad.responsive_search_ad.path2.value = 'deals'
+
+    try:
```

Comment:
Nit: Let's add comments similar to the above like "Send a request to the server to add a responsive search ad"

Reply:
fixed.